domain/relation/watcher_test.go

```diff
@@ -108,6 +116,214 @@ AND key = 'key'
+	svc := s.setupService(c, factory)
+	watcher, err := svc.WatchLifeSuspendedStatus(context.Background(), unitUUID)
+	c.Assert(err, jc.ErrorIsNil)
+
+	relationKey := relationtesting.GenNewKey(c, "two:fake-provides my-application:fake-provides").String()
+	harness := watchertest.NewHarness(s, watchertest.NewWatcherC(c, watcher))
+
+	// Act: change the relation life.
+	harness.AddTest(func(c *gc.C) {
+		err := s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+			if _, err := tx.ExecContext(ctx, "UPDATE relation SET life_id = 1 WHERE uuid=?", relationUUID); err != nil {
+				return errors.Capture(err)
+			}
+			return nil
+		})
+		c.Assert(err, jc.ErrorIsNil)
+	}, func(w watchertest.WatcherC[[]string]) {
+		// Assert: received changed of relation key.
+		w.Check(
+			watchertest.StringSliceAssert[string](relationKey),
+		)
+	})
+
+	// Act: change the relation status other than suspended.
+	harness.AddTest(func(c *gc.C) {
+		err := s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+			if _, err := tx.ExecContext(ctx, "UPDATE relation_status SET relation_status_type_id = 3 WHERE relation_uuid=?", relationUUID); err != nil {
+				return errors.Capture(err)
+			}
+			return nil
+		})
+		c.Assert(err, jc.ErrorIsNil)
+	}, func(w watchertest.WatcherC[[]string]) {
+		// Assert: no change received. Change received only if status changes to
+		// suspended.
+		w.AssertNoChange()
+	})
+
+	// Act: change the relation status to suspended.
+	harness.AddTest(func(c *gc.C) {
+		err := s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+			if _, err := tx.ExecContext(ctx, "UPDATE relation_status SET relation_status_type_id = 4 WHERE relation_uuid=?", relationUUID); err != nil {
+				return errors.Capture(err)
+			}
+			return nil
+		})
+		c.Assert(err, jc.ErrorIsNil)
+	}, func(w watchertest.WatcherC[[]string]) {
+		// Assert: received changed of relation key, relation status changed to suspended.
+		w.Check(
+			watchertest.StringSliceAssert[string](relationKey),
+		)
+	})
+
+	// Act: change the relation status to joined and life to dead, to get
+	// changes on both tables watched.
+	harness.AddTest(func(c *gc.C) {
+		err := s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+			if _, err := tx.ExecContext(ctx, "UPDATE relation SET life_id = 2 WHERE uuid=?", relationUUID); err != nil {
+				return errors.Capture(err)
+			}
+			if _, err := tx.ExecContext(ctx, "UPDATE relation_status SET relation_status_type_id = 1 WHERE relation_uuid=?", relationUUID); err != nil {
+				return errors.Capture(err)
+			}
+			return nil
+		})
+		c.Assert(err, jc.ErrorIsNil)
+	}, func(w watchertest.WatcherC[[]string]) {
+		// Assert: with changes in both tables at the same time, the relation
+		// key is sent once.
+		w.Check(
+			watchertest.StringSliceAssert[string](relationKey),
+		)
+	})
+
+	harness.Run(c, []string{relationKey})
+}
```

Comment:
It would be good to exercise a bit of the `ignoredRelations` codepaths, possibily in another test, but checking that when it encounters a new relation that is not meant to be watched it correctly ignores it multiple times in a row.

Reply:
done